companion/gateway/utils/query_executor.py

```
@@ -0,0 +1,375 @@
+        """
+        try:
+            parsed = sqlparse.parse(query)
+            if not parsed:
+                return False
+
+            # Check each statement (handles multi-statement queries)
+            for statement in parsed:
+                if QueryExecutor._check_user_id_in_statement(statement):
+                    return True
+
+            return False
```

Ruff (0.13.2)

Consider moving this statement to an `else` block

(TRY300)